test/analysis/headerCheck/CMakeLists.txt

```diff
@@ -96,9 +99,12 @@ ALPAKA_ADD_EXECUTABLE(
 TARGET_COMPILE_OPTIONS(
     ${_TARGET_NAME}
     PRIVATE ${_DEFINITIONS_PRIVATE})
+TARGET_INCLUDE_DIRECTORIES(
+    ${_TARGET_NAME}
+    PRIVATE ${_INCLUDE_DIRECTORIES_PRIVATE})
 TARGET_LINK_LIBRARIES(
     ${_TARGET_NAME}
-    PUBLIC common Boost::unit_test_framework)
+    PRIVATE common ${_LINK_LIBRARIES_PRIVATE})
```

Comment:
maybe with `if(TARGET Boost::unit_test_framework)` ? This would still in most cases give the more stable target to link...